lib/IDE/CodeCompletion.cpp

```diff
@@ -5930,12 +5933,13 @@ void CodeCompletionCallbacksImpl::doneParsing() {
     if (!OnRoot && KPE->getComponents().back().getKind() ==
                        KeyPathExpr::Component::Kind::OptionalWrap) {
       // KeyPath expr with '?' (e.g. '\Ty.[0].prop?.another').
-      // Althogh expected type is optional, we should unwrap it because it's
+      // Although expected type is optional, we should unwrap it because it's
       // unwrapped.
       baseType = baseType->getOptionalObjectType();
     }
 
-    Lookup.getValueExprCompletions(baseType);
+    Lookup.getValueExprCompletions(baseType, /*VD=*/nullptr,
+                                   /*allowEraseOptionalLeadingDot*/ !OnRoot);
```

Comment:
Instead of passing an extra argument, can't we use `IsAfterSwiftKeyPathRoot` flag?

Reply:
Sure! I just didn't know this flag

Reply:
Fixed!

Reply:
Great! Thank you!